clojure-ts-mode.el

```diff
@@ -2475,20 +2461,28 @@ parenthesis."
 
 (defconst clojure-ts-grammar-recipes
   '((clojure "https://github.com/sogaiu/tree-sitter-clojure.git"
-             "v0.0.13")
+             "unstable-20250526")
     (markdown-inline "https://github.com/MDeiml/tree-sitter-markdown"
                      "v0.4.1"
                      "tree-sitter-markdown-inline/src")
     (regex "https://github.com/tree-sitter/tree-sitter-regex"
            "v0.24.3"))
   "Intended to be used as the value for `treesit-language-source-alist'.")
 
+(defun clojure-ts--grammar-outdated-p ()
+  "Return TRUE if currently installed grammar is outdated."
+  (treesit-query-valid-p 'clojure '((sym_lit (meta_lit)))))
+
 (defun clojure-ts--ensure-grammars ()
   "Install required language grammars if not already available."
   (when clojure-ts-ensure-grammars
     (dolist (recipe clojure-ts-grammar-recipes)
       (let ((grammar (car recipe)))
-        (unless (treesit-language-available-p grammar nil)
+        (when (or (not (treesit-language-available-p grammar nil))
```

Comment:
I'd probably push all this logic to the `outdated` function, as it seems related.

Reply:
This also has to be checked for markdown and regex grammars, so it belongs here.

Reply:
I was mostly referring to the check `(equal grammar 'clojure)`. It also seemed to me it's the only one doing some outdated check.

Reply:
I can either add a `grammar` parameter to `clojure-ts--clojure-grammar-outdated-p` (which IMO would look weird), or leave this check here as is.

The `grammar` here can be one of three values: clojure, markdown-inline or regex. If I just check that Clojure grammar is outdated when we're trying to install markdown-inline grammar it might re-install markdown-grammar too, which is not the desired outcome.

Reply:
It's not a big deal. Let's leave it as is.